core/log/profiler_hook.cpp

```diff
@@ -1,23 +1,72 @@
-// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+// SPDX-FileCopyrightText: 2017 - 2025 The Ginkgo authors
 //
 // SPDX-License-Identifier: BSD-3-Clause
 
 #include "ginkgo/core/log/profiler_hook.hpp"
 
+#include <complex>
 #include <memory>
 #include <mutex>
 #include <sstream>
 
 #include <ginkgo/core/base/name_demangling.hpp>
+#include <ginkgo/core/config/config.hpp>
+#include <ginkgo/core/distributed/vector.hpp>
 #include <ginkgo/core/log/logger.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
 #include <ginkgo/core/solver/solver_base.hpp>
 #include <ginkgo/core/stop/criterion.hpp>
 
+#include "core/base/dispatch_helper.hpp"
+#include "core/config/type_descriptor_helper.hpp"
 #include "core/log/profiler_hook.hpp"
 
-
 namespace gko {
 namespace log {
+namespace {
+
+
+std::string check_vector_type(const LinOp* linop)
```

Comment:
Perhaps this should better be merged with `stringify_objects`. If a user explicitly sets a name for an object, it should override this behavior here.

Reply:
By the way, isn't `stringify_objects` already enough?